File: problem_2/problem2.rb
def solution(limit)
  fib_seq        = [1, 2]
  sum_even_terms = 0

  loop do
    if fib_seq[-1] % 2 == 0 then sum_even_terms += fib_seq[-1] end
    
    new_term = fib_seq[-2] + fib_seq[-1]
    break if fib_seq[-1] > limit
    fib_seq << new_term
  end
  puts sum_even_terms
end

solution(4000000)

File: problem_2/problem2.js
var solution = function() {
  var LIMIT   = 4000000;
  var fibSeq = [1, 2];
  var sumEvenTerms = 0;

  // Populate the Fibonacci array
  while (fibSeq[fibSeq.length - 1] < LIMIT) {
    newTerm = fibSeq[fibSeq.length - 2] + fibSeq[fibSeq.length - 1];
    fibSeq.push(newTerm);
  }
  
  // Get the sum of all the even numbers in the sequence
  for (i = 0; i < fibSeq.length; i++) {
    if (fibSeq[i] % 2 === 0) {
      sumEvenTerms += fibSeq[i];
    }
  }

  console.log(sumEvenTerms);
};

solution();